File: bricks/pac_native_plugin/__brick__/packages/{{name.snakeCase()}}/android/src/main/kotlin/com/danhdue/{{name.snakeCase()}}/data/repository/{{name.pascalCase()}}RepositoryImpl.kt
/*
 * Copyright (c) 2026, one of DanhDue ExOICTIF projects. All rights reserved.
 */

package com.danhdue.{{name.snakeCase()}}.data.repository

import com.danhdue.{{name.snakeCase()}}.domain.model.{{name.pascalCase()}}Data
import com.danhdue.{{name.snakeCase()}}.domain.repository.{{name.pascalCase()}}Repository
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class {{name.pascalCase()}}RepositoryImpl @Inject constructor() : {{name.pascalCase()}}Repository {
    override suspend fun getData(): Result<{{name.pascalCase()}}Data> {
        return Result.success(
            {{name.pascalCase()}}Data(
                id = "sample-data-1",
                title = "Native Android {{name.pascalCase()}} Payload"
            )
        )
    }

    override suspend fun syncData(): Result<Boolean> {
        return Result.success(true)
    }
}

File: bricks/pac_native_plugin/__brick__/packages/{{name.snakeCase()}}/ios/{{name.snakeCase()}}/Sources/{{name.snakeCase()}}/Data/Repository/{{name.pascalCase()}}RepositoryImpl.swift
// Copyright (c) 2026, one of DanhDue ExOICTIF projects. All rights reserved.

import Foundation

/// Default implementation of {{name.pascalCase()}}Repository as an actor.
actor {{name.pascalCase()}}RepositoryImpl: {{name.pascalCase()}}Repository {
    private var currentData: {{name.pascalCase()}}Data

    init(initialData: {{name.pascalCase()}}Data? = nil) {
        currentData = initialData ?? {{name.pascalCase()}}Data(id: "default-id", title: "Native {{name.pascalCase()}} Data")
    }

    func getData() async throws -> {{name.pascalCase()}}Data {
        currentData
    }

    func syncData() async throws {
        currentData = {{name.pascalCase()}}Data(id: UUID().uuidString, title: "Synced {{name.pascalCase()}} Data", timestamp: Date())
    }
}
